src/pybind/mgr/dashboard/services/cephfs.py

```diff
@@ -260,3 +260,46 @@ def set_quotas(self, path, max_bytes=None, max_files=None):
         if max_files is not None:
             self.cfs.setxattr(path, 'ceph.quota.max_files',
                               str(max_files).encode(), 0)
+
+    def try_write(self, path, buf):
+        """
+        Try write some data to the specified path.
+        :param path: The path of the directory/file.
+        :type path: str.
+        :param buf: The str to write to the buf.
+        :type buf: str.
+        """
+        try:
+            fd = self.cfs.open(path, 'w', 644)
+        except cephfs.Error:
+            if fd is not None:
+                self.cfs.close(fd)
+        try:
+            self.cfs.write(fd, buf.encode('utf-8'), 0)
+        except cephfs.Error as e:
+            logger.debug("EIO: %s", str(e))
+        finally:
+            self.cfs.close(fd)
```

Comment:
```suggestion
        try:
            fd = self.cfs.open(path, 'w', 644)
            self.cfs.write(fd, buf.encode('utf-8'), 0)
        except cephfs.Error as e:
            logger.debug("EIO: %s", str(e))
        finally:
            self.cfs.close(fd)
```

Reply:
got it

Reply:
more better thing to do would be to have finally block like this:
```
finally:
    if fd is not None:
        self.cfs.close(fd)
```
this would make sure we dont close the file if it was never opened(like error at line `fd = self.cfs.open(path, 'w', 644)`)

Reply:
updated